mindmeld/active_learning/alp.py

```diff
@@ -0,0 +1,234 @@
+from copy import deepcopy
+
+import logging
+
+import math
+from .data_loading import DataBucketFactory
+from .output_manager import OutputManager
+from .plot_manager import PlotManager
+from .classifiers import MindMeldClassifier
+from .heuristics import HeuristicsFactory
+from ..constants import STRATEGY_ABRIDGED
+
+logger = logging.getLogger(__name__)
+
+
+class ActiveLearningPipeline:  # pylint: disable=R0902
+    """Class that executes the training and selection process for the Active Learning Pipeline"""
+
+    def __init__(  # pylint: disable=R0913
+        self,
+        app_path: str,
+        train_pattern: str,
+        test_pattern: str,
+        load: bool,
+        save: bool,
+        train_seed_pct: float,
+        n_classifiers: int,
+        n_epochs: int,
+        batch_size: int,
+        training_strategies: list,
+        training_level: str,
+        log_selection_strategy: str,
+        save_sampled_queries: bool,
+        early_stopping_window: int,
+        log_usage_pct: float,
+        labeled_logs_pattern: str,
+        unlabeled_logs_path: str,
+        output_folder: str,
+    ):
+        """
+        Args:
+            app_path (str): Path to MindMeld application
+            train_pattern (str): Regex pattern to match train files. For example, ".*train.*.txt"
+            test_pattern (str): Regex pattern to match test files. For example, ".*test.*.txt"
+            load (bool): Whether to load pickled queries from a local folder
+            save (bool): Whether to save queries as a local pickle file
+            train_seed_pct (float): Percentage of training data to use as the initial seed
+            n_classifiers (int): Number of classifiers to be used by a subset of heuristics
+            n_epochs (int): Number of epochs to run training
+            batch_size (int): Number of queries to select at each iteration
+            training_level (str): The hierarchy level to train ("domain" or "intent")
+            training_strategies (List[str]): List of strategies to use for training
+            log_selection_strategy (str): Single strategy to use for log selection
```

Comment:
Is there a need to say `log selection` vs just `selection`?

Reply:
The intention was to try to make this param more distinct but I think you're right in that we can just use "selection". Updated accordingly